pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
+
+        return self.special_set
+    
+    def update_values(self, values):
         """
-        Generate a dictionary, keys are the variable names, values are the indexes of this variable.
-        For e.g., name_and_index = {'C': ['CA', 'CB', 'CC']}
+        Used for updating values
+
         Parameters
         ---------
-        all_info: a dictionary, keys are measurement variable names,
-                values are a dictionary, keys are its extra index, values are its measuring time points
-                values are a list of measuring time point if there is no extra index for this measurement
-            Note: all_info can be the self.measurement_all_info, but does not have to be it.
+         values: a ``list`` containing values which has the same shape of flattened variables 
+            default choice is None, means there is no give nvalues 
         """
-        measurement_name = list(all_info.keys())
-        # a list of measurement extra indexes
-        measurement_extra_index = []
-        # check if the measurement has extra indexes
-        for i in measurement_name:
-            if type(all_info[i]) is dict:
-                index_list = list(all_info[i].keys())
-                measurement_extra_index.append(index_list)
-            elif type(all_info[i]) is list:
-                measurement_extra_index.append(None)
-        # a dictionary, keys are measurement names, values are a list of extra indexes
-        self.name_and_index = dict(zip(measurement_name, measurement_extra_index))
-
-    def _generate_flatten_name(self, measure_name_and_index):
+        self.special_set_value = self._generate_dict(values)
+        
+
+    def _generate_dict(self, values):
         """
-        Generate measurement flattened names
-        Parameters
-        ----------
-        measure_name_and_index: a dictionary, keys are measurement names, values are lists of extra indexes
+        Given a list of values, return a dictionary, keys are special set names. 
+        """
+        value_map = {}
+        for i in range(len(values)):
+            value_map[self.special_set[i]] = values[i]
 
-        Returns
-        ------
-        jac_involved_name: a list of flattened measurement names
+        return value_map
+
+
+    def _add_elements(self, var_name, extra_index=None, time_index=None):
         """
-        flatten_names = []
-        for j in measure_name_and_index.keys():
-            if measure_name_and_index[j] is not None: # if it has extra index
-                for ind in measure_name_and_index[j]:
-                    flatten_name = j + self.ind_string + str(ind)
-                    flatten_names.append(flatten_name)
-            else:
-                flatten_names.append(j)
-
-        self.flatten_measure_name = flatten_names
-
-    def _generate_variance(self, flatten_measure_name, variance, name_and_index):
+        Used for generating string names with indexes. 
+
+        Parameter 
```

Comment:
See my comments in the examples for how to refactor the API. This method will be impacted the most.